dataset/src/test/scala/frameless/CreateTests.scala

```diff
@@ -28,6 +28,66 @@ class CreateTests extends TypedDatasetSuite with Matchers {
       Vector[(Food, Country)]] _))
   }
 
+  test("array fields") {
+
+    def prop[T](implicit arb: Arbitrary[Array[T]], encoder: TypedEncoder[X1[Array[T]]]) = forAll {
+      data: Array[T] =>
+        val Seq(X1(arr)) = TypedDataset.create(Seq(X1(data))).collect().run()
```

Comment:
What about a test where `Array` is root element, not field, so type would be `TypedDataset[Array[T]]`